### DISCUSSION TITLE
problem with custom button in inspector

### DISCUSSION THREAD

**Discussion by PFlorient** [clientIO/joint#2373] [Q&A] [answered]
### Introduction

I've just created a button to copy and paste my current cell. However I'm encountering a problem. When I click on my button, my copy is made only once.

### Steps to reproduce

To do this, I've used `renderFieldContent()` and inside 

```
const $buttonSet = $('<div/>');
const $copy = $('<button><img src="assets/icons/copy.svg"/></button>')
$buttonSet.append([$copy])
$copy.on('click', () => {
   $buttonSet.data('utils', 'copy');
    inspector.updateCell($buttonSet, path, options)
})
return $buttonSet;
```

 I put some logs in 
`graph.on('change:attrs', (cell, evt, inspector) => {...});`
but here the log is only displayed on the first copy (and doesn't work on the duplicated element)

(I have the same problem on other elements that have nothing to do with it (just a `set` for a cell))

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
I don't see where the `attrs` being changed. Could you please provide more complete code?
Please note that `change:attrs` is not triggered when there was nothing changed (the current and the new value is the same).

**Reply by PFlorient** [reply]
i'm not looking to change the `attrs` directly
It's by doing a graph.on('all') that I've intercepted the different events that I've noticed that clicking on my button triggers this event. So I based myself on it to manage what I need. 
If this isn't the right way, what is? 
I based myself on this part of the documentation: https://resources.jointjs.com/docs/rappid/v3.3/ui.html#ui.Inspector.custom-fields

**Reply by kumilingus** [reply, marked as answer]
I am not sure I follow.

You can call the action directly.
```js
$copy.on('click', () => myAction())
```

You can trigger an event e.g on the inspector.
```js
const inspector = new ui.Inspector({
   /* ... */
   renderFieldContent() {
     /* ... */
     $copy.on('click', () => inspector.trigger('my-action'))
  }
});

inspector.on('my-action', () => myAction());
```



```
